mqtt/mqttd/src/broker/bootstrap/edgehub.rs

```diff
@@ -165,50 +165,35 @@ async fn server_certificate_renewal(renew_at: DateTime<Utc>) {
 async fn start_sidecars(
     broker_handle: BrokerHandle,
     system_address: String,
-) -> (SidecarShutdownHandle, JoinHandle<Result<()>>) {
-    let (termination_handle, tx) = channel::<()>();
+) -> Result<(SidecarShutdownHandle, JoinHandle<Result<()>>)> {
+    let (sidecar_termination_handle, sidecar_termination_receiver) = channel::<()>();
 
-    let event_loop = tokio::spawn(async move {
-        info!("starting command handler...");
-        let (mut command_handler_shutdown_handle, command_handler_join_handle) =
-            start_command_handler(broker_handle, system_address).await?;
+    let device_id = env::var(DEVICE_ID_ENV)?;
+    let command_handler =
+        CommandHandler::new(broker_handle, system_address, device_id.as_str()).await?;
+    let command_handler_shutdown_handle = command_handler.shutdown_handle()?;
 
-        start_bridge().await?;
+    let mut bridge_controller = BridgeController::new();
 
-        tx.await?;
+    let event_loop = tokio::spawn(async move {
+        let command_handler_join_handle = tokio::spawn(command_handler.run());
 
-        command_handler_shutdown_handle.shutdown().await?;
-        command_handler_join_handle.await??;
-        info!("command handler shutdown.");
-        Ok::<(), anyhow::Error>(())
-    });
+        let bridge = bridge_controller.start();
 
-    (SidecarShutdownHandle(termination_handle), event_loop)
-}
+        sidecar_termination_receiver.await?;
 
-async fn start_command_handler(
-    broker_handle: BrokerHandle,
-    system_address: String,
-) -> Result<(
-    CommandShutdownHandle,
-    JoinHandle<Result<(), CommandHandlerError>>,
-)> {
-    let device_id = env::var(DEVICE_ID_ENV)?;
-    let command_handler = CommandHandler::new(broker_handle, system_address, device_id.as_str());
-    let shutdown_handle = command_handler.shutdown_handle()?;
-
-    let join_handle = tokio::spawn(command_handler.run());
+        command_handler_shutdown_handle.shutdown().await?;
+        command_handler_join_handle.await?;
 
-    Ok((shutdown_handle, join_handle))
-}
+        bridge.await?;
```

Comment:
As I understand, bridge shutdown is not yet implemented?

Reply:
Yep. When it gets implemented then we can add the shutdown logic like we do here.